fvtest/compilertriltest/CompareTest.cpp

```diff
@@ -95,3 +95,64 @@ INSTANTIATE_TEST_CASE_P(CompareTest, Int32Compare, ::testing::Combine(
         std::make_tuple("icmpge", icmpge),
         std::make_tuple("icmplt", icmplt),
         std::make_tuple("icmple", icmple) )));
+
+int32_t lcmpeq(int64_t l, int64_t r) {
+    return (l == r) ? 1 : 0;
+}
+
+int32_t lcmpne(int64_t l, int64_t r) {
+    return (l != r) ? 1 : 0;
+}
+
+int32_t lcmpgt(int64_t l, int64_t r) {
+    return (l > r) ? 1 : 0;
+}
+
+int32_t lcmpge(int64_t l, int64_t r) {
+    return (l >= r) ? 1 : 0;
+}
+
+int32_t lcmplt(int64_t l, int64_t r) {
+    return (l < r) ? 1 : 0;
+}
+
+int32_t lcmple(int64_t l, int64_t r) {
+    return (l <= r) ? 1 : 0;
+}
+
+int32_t lcmp(int64_t l, int64_t r) {
+    return (l < r) ? -1 : ((l > r) ? 1 : 0);
+}
+
+class Int64Compare : public TRTest::OpCodeTest<int32_t,int64_t,int64_t> {};
+
+TEST_P(Int64Compare, UsingLoadParam) {
```

Comment:
Any particular reason for not having a `UsingConst` variant of this test?

Reply:
I didn't include it this time because some of the tests in the UsingConst version failed on x86 Linux like this:
```
[----------] 289 tests from CompareTest/Int64Compare
/path/to/CompareTest.cpp:148: Failure
      Expected: exp
      Which is: 1
To be equal to: act
      Which is: 0
[  FAILED  ] CompareTest/Int64Compare.UsingConst/5, where GetParam() = ((0, -9223372036854775808), ("lcmpne", 0x7947b0)) (5 ms)
/path/to/CompareTest.cpp:148: Failure
      Expected: exp
      Which is: 1
To be equal to: act
      Which is: 0
[  FAILED  ] CompareTest/Int64Compare.UsingConst/24, where GetParam() = ((1, -9223372036854775807), ("lcmpne", 0x7947b0)) (4 ms)
/path/to/CompareTest.cpp:148: Failure
      Expected: exp
      Which is: 1
To be equal to: act
      Which is: 0
[  FAILED  ] CompareTest/Int64Compare.UsingConst/40, where GetParam() = ((-1, 9223372036854775807), ("lcmpne", 0x7947b0)) (4 ms)
```
I need to investigate what is wrong with it.

Reply:
Alright. Thanks @knn-k.

If you can't figure it out quickly, just add it as a disabled test (add `DISABLED_` prefix to the test name) and open an issue about it. I strongly suspect TreeSimplifier is folding the constants incorrectly. Do you have a trace log?